internal/controllers/topology/cluster/patches/variables/variables_test.go

```diff
@@ -66,12 +66,135 @@ func TestGlobal(t *testing.T) {
 						Class:   "clusterClass1",
 						Version: "v1.21.1",
 					},
+					ClusterNetwork: &clusterv1.ClusterNetwork{
+						Services: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"10.10.10.1/24"},
+						},
+						Pods: &clusterv1.NetworkRanges{
+							CIDRBlocks: []string{"11.10.10.1/24"},
+						},
+						ServiceDomain: "cluster.local",
+					},
 				},
 			},
 			want: VariableMap{
-				"location":   toJSON("\"us-central\""),
-				"cpu":        toJSON("8"),
-				BuiltinsName: toJSON("{\"cluster\":{\"name\":\"cluster1\",\"namespace\":\"default\",\"topology\":{\"version\":\"v1.21.1\",\"class\":\"clusterClass1\"}}}"),
+				"location": toJSON("\"us-central\""),
+				"cpu":      toJSON("8"),
+				BuiltinsName: toJSONCompact(`{
+"cluster":{
+  "name": "cluster1",
+  "namespace": "default",
+  "topology":{
+    "version": "v1.21.1",
+    "class": "clusterClass1"
+  },
+  "network":{
+    "serviceDomain":"cluster.local",
+    "services":["10.10.10.1/24"],
+    "pods":["11.10.10.1/24"],
+    "ipFamily": "IPv4IPFamily"
```

Comment:
Q: Should we make this "IPv4" instead? I understand why the go types have that suffix, but I think in general I think it's not good if the actual enum values have the enum type as suffix 
```suggestion
    "ipFamily": "IPv4"
```

Reply:
On the other side that's what the IPFamily func returns, but I think it would be really not good to have those suffixes.


@vincepri @fabriziopandini Opinions?

Reply:
We're not doing any validation in this PR - these are the types defined in the api package under Cluster types. For the test I can change it to "foo" if we want. Is there need to do some specific evaluation or transformation of these types in this PR?

Reply:
Oh sorry for the confusion. I just added the comment at the test because it seemed convenient. I wanted to discuss to what values we should set that builtin variable.

Reply:
If we want to change the return value of IPFamily func we could:
* set the builtin variables without suffix
* add a “deprecation note” in the migration doc that with the next CAPI minor release (v1.3) this func will return the values without suffix

I think this func is barely used by anyone: https://grep.app/search?q=GetIPFamily&filter[lang][0]=Go

Reply:
Given that we document IPFamily as 
// IPFamily is the IPFamily the Cluster is operating in. One of Invalid, IPv4, IPv6, DualStack.

And that we have to backport, I would simply drop the suffix

Reply:
@fabriziopandini just to be clear - I should drop the suffix in the existing function?

Reply:
The API cannot be changed, otherwise it will be a breaking change.
However, we can drop the suffix generated by the API when setting the built-in variable

Reply:
So like this: https://github.com/kubernetes-sigs/cluster-api/blob/8dec3d73b743c36690474d096e9c99330f227bb8/api/v1alpha4/cluster_types.go#L384

Just updated now :smile:

Reply:
I think Fabrizio meant transforming the strings returned by the func in the variable code (or alternatively something with switch case, which is probably more stable)

This change is currently a breaking change. (and in the v1alpha4 package :) )

Reply:
Upadated, but I think we might be better off deprecating and adding the method in the API instead of creating 2 .toString methods for this enum.

Reply:
@killianmuldoon thanks for updating
we can tackle API deprecation in a separated PR, but I would like to keep this as small as possible for enabling backporting

Reply:
Sure thing! I'll open an additional PR and we can discuss there.

Reply:
Yup good point, we can't really cherry-pick deprecations.